cuda_phase-retrieval_class/test/test_8_cupy_reference_cuda_interface/test_interface.py

```diff
@@ -44,8 +44,26 @@ def test_add_cupy_and_numpy():
     for i in range(3):
         c[i] = data[i]
 
-    assert(cp.array_equal(c, c_inc))
+    assert cp.array_equal(c, c_inc)
 
 
+def test_memory_sychronization():
+    """
+    Check if automatic copy to host and copy to device are correct done,
+    if c++ cuda kernel is called via python binding
+    """
+    size = 30
 
+    mem_holder = Test_Interface.GPU_memory_holder([size])
+    mem_holder_ref = mem_holder.get_memory_reference()
+    mem_holder_array = cp.array(mem_holder_ref, dtype=mem_holder_ref.dtype, copy=False)
```

Comment:
Please develop a member function in cupy_ref, which returns the cupy array:

`mem_holder_array = mem_holder_ref.get_cupy_array()`

Reply:
I think I need to add one more parameter to cupy_ref.py, because my "typestr" for __cuda_array_interface__ is not implemented properly.

Reply:
Okay, than do it.

Reply:
Why?

Reply:
Because I think can simply use the private member 
```C++
Cupy_Ref<T> gpu_memory_reference
```

Reply:
But the main task of the PR is to create an interface to get easily a cupy array from every cupy ref.

Reply:
okay